### PR TITLE
Manual updates 20240509

### PR DESCRIPTION
**Pull request by moljac**
### Describe your contribution

Updates:

- `com.google.android.gms:play-services-auth` - 21.0.0 -> 21.1.1
- `com.google.android.gms:play-services-authbase` - 18.0.11 -> 18.0.12
- `com.google.android.gms:play-services-base` - 18.3.0 -> 18.4.0
- `com.google.android.gms:play-services-cloudmessaging` - 17.1.0 -> 17.2.0
- `com.google.android.gms:play-services-fido` - 20.1.0 -> 21.0.0
- `com.google.android.gms:play-services-gamesv2` - 19.0.0 -> 20.0.0
- `com.google.android.gms:play-services-location` - 21.1.0 -> 21.2.0
- `com.google.android.gms:play-services-nearby` - 19.1.0 -> 19.2.0
- `com.google.android.gms:play-services-wallet` - 19.2.1 -> 19.3.0
- `com.google.android.play:asset-delivery` - 2.1.0 -> 2.2.2
- `com.google.android.play:asset-delivery-ktx` - 2.1.0 -> 2.2.2
- `com.google.firebase:firebase-abt` - 21.1.2 -> 22.0.0
- `com.google.firebase:firebase-appcheck` - 17.1.2 -> 18.0.0
- `com.google.firebase:firebase-appcheck-debug` - 17.1.2 -> 18.0.0
- `com.google.firebase:firebase-appcheck-ktx` - 17.1.2 -> 18.0.0
- `com.google.firebase:firebase-appcheck-play-integrity` - 17.1.2 -> 18.0.0
- `com.google.firebase:firebase-common` - 20.4.2 -> 21.0.0
- `com.google.firebase:firebase-common-ktx` - 20.4.2 -> 21.0.0
- `com.google.firebase:firebase-components` - 17.1.5 -> 18.0.0
- `com.google.firebase:firebase-config` - 21.6.1 -> 22.0.0
- `com.google.firebase:firebase-config-ktx` - 21.6.1 -> 22.0.0
- `com.google.firebase:firebase-crashlytics` - 18.6.2 -> 19.0.0
- `com.google.firebase:firebase-crashlytics-ktx` - 18.6.2 -> 19.0.0
- `com.google.firebase:firebase-crashlytics-ndk` - 18.6.2 -> 19.0.0
- `com.google.firebase:firebase-database` - 20.3.0 -> 21.0.0
- `com.google.firebase:firebase-database-ktx` - 20.3.0 -> 21.0.0
- `com.google.firebase:firebase-datatransport` - 18.2.1 -> 19.0.0
- `com.google.firebase:firebase-dynamic-links` - 21.2.0 -> 22.0.0
- `com.google.firebase:firebase-dynamic-links-ktx` - 21.2.0 -> 22.0.0
- `com.google.firebase:firebase-firestore` - 24.10.2 -> 25.0.0
- `com.google.firebase:firebase-firestore-ktx` - 24.10.2 -> 25.0.0
- `com.google.firebase:firebase-functions` - 20.4.0 -> 21.0.0
- `com.google.firebase:firebase-functions-ktx` - 20.4.0 -> 21.0.0
- `com.google.firebase:firebase-inappmessaging` - 20.4.0 -> 21.0.0
- `com.google.firebase:firebase-inappmessaging-display` - 20.4.0 -> 21.0.0
- `com.google.firebase:firebase-inappmessaging-display-ktx` - 20.4.0 -> 21.0.0
- `com.google.firebase:firebase-inappmessaging-ktx` - 20.4.0 -> 21.0.0
- `com.google.firebase:firebase-installations` - 17.2.0 -> 18.0.0
- `com.google.firebase:firebase-installations-ktx` - 17.2.0 -> 18.0.0
- `com.google.firebase:firebase-messaging` - 23.4.1 -> 24.0.0
- `com.google.firebase:firebase-messaging-directboot` - 23.4.1 -> 24.0.0
- `com.google.firebase:firebase-messaging-ktx` - 23.4.1 -> 24.0.0
- `com.google.firebase:firebase-perf` - 20.5.2 -> 21.0.0
- `com.google.firebase:firebase-perf-ktx` - 20.5.2 -> 21.0.0
- `com.google.firebase:firebase-sessions` - 1.2.2 -> 2.0.0
- `com.google.firebase:firebase-storage` - 20.3.0 -> 21.0.0
- `com.google.firebase:firebase-storage-ktx` - 20.3.0 -> 21.0.0
- `com.google.mlkit:facedetection` - 16.1.5 -> 16.1.6
- `com.google.mlkit:imagelabeling` - 17.0.7 -> 17.0.8
- `com.google.mlkit:languageid` - 17.0.4 -> 17.0.5
- `com.google.mlkit:objectdetection` - 17.0.0 -> 17.0.1
- `com.google.mlkit:smartreply` - 17.0.2 -> 17.0.3
- `com.google.android.libraries.places:places` - 3.3.0 -> 3.4.0
- `com.android.billingclient:billing` - 6.1.0 -> 6.2.1
- `com.android.billingclient:billing-ktx` - 6.1.0 -> 6.2.1
- `com.google.android.datatransport:transport-api` - 3.1.0 -> 3.2.0
- `com.google.android.datatransport:transport-backend-cct` - 3.2.0 -> 3.3.0
- `com.google.android.datatransport:transport-runtime` - 3.2.0 -> 3.3.0
- `com.google.dagger:dagger` - 2.49 -> 2.50
- `com.google.errorprone:error_prone_annotations` - 2.26.1 -> 2.27.1
- `com.google.flatbuffers:flatbuffers-java` - 23.5.26 -> 24.3.25
- `com.google.protobuf:protobufjavalite` - 3.25.3 -> 4.26.1
- `com.squareup.okio:okio` - 3.8.0 -> 3.9.0
- `com.squareup.okio:okio-jvm` - 3.8.0 -> 3.9.0
- `com.squareup.retrofit2:adapter-rxjava2` - 2.9.0 -> 2.10.0
- `com.squareup.retrofit2:converter-gson` - 2.9.0 -> 2.10.0
- `com.squareup.retrofit2:converter-scalars` - 2.9.0 -> 2.10.0
- `com.squareup.retrofit2:retrofit` - 2.9.0 -> 2.10.0
- `io.grpc:grpc-android` - 1.58.1 -> 1.59.1
- `io.grpc:grpc-api` - 1.58.1 -> 1.59.1
- `io.grpc:grpc-context` - 1.58.1 -> 1.59.1
- `io.grpc:grpc-core` - 1.58.1 -> 1.59.1
- `io.grpc:grpc-okhttp` - 1.58.1 -> 1.59.1
- `io.grpc:grpc-protobuf-lite` - 1.58.1 -> 1.59.1
- `io.grpc:grpc-stub` - 1.58.1 -> 1.59.1
- `io.grpc:grpc-util` - 1.58.1 -> 1.59.1
- `org.tensorflow:tensorflow-lite` - 2.15.0 -> 2.16.1
- `org.tensorflow:tensorflow-lite-api` - 2.15.0 -> 2.16.1
- `org.tensorflow:tensorflow-lite-gpu` - 2.15.0 -> 2.16.1
- `org.tensorflow:tensorflow-lite-gpu-api` - 2.15.0 -> 2.16.1
- `org.tensorflow:tensorflow-lite-select-tfops` - 2.15.0 -> 2.16.1
- `com.google.inject:guice` - 5.1.0 -> 6.0.0
- `org.ow2.asm:asm` - 9.2 -> 9.5
- `jakarta.inject:jakarta.injectapi` -  -> 2.0.1
- `androidx.lifecycle:lifecycle-extensions` -  -> 2.2.0
- `com.google.android.gms:playservices-ads-base` -  -> 20.0.0
- `com.google.android.gms:playservices-measurement-base` -  -> 16.3.0

### CODE REVIEW

**Comment by moljac** [comment]
Unable to update due to preview dependencies

```
com.google.android.gms:play-services-ads*
  androidx.privacysandbox.ads:ads-adservices:1.0.0-beta05
  androidx.privacysandbox.ads:ads-adservices-java:1.0.0-beta05

com.google.android.gms:play-services-measurement*
  androidx.privacysandbox.ads:ads-adservices:1.0.0-beta05
  androidx.privacysandbox.ads:ads-adservices-java:1.0.0-beta05

com.google.ads.interactivemedia.v3:interactivemedia
    com.google.android.tv.tv-ads:1.0.0-alpha02
```

File: config.json [hunk 1]
```diff
@@ -1962,10 +1962,10 @@
       },
       {
         "groupId": "com.squareup.okhttp3",
-        "artifactId": "okhttp",
+        "artifactId": "okhttp-brotli",
         "version": "4.12.0",
-        "nugetVersion": "4.12.0.4",
-        "nugetId": "Square.OkHttp3",
+        "nugetVersion": "4.12.0.1",
```
Comment (jpobst) [inline config.json:1967]: Downgrade from existing `4.12.0.2`.

File: config.json [hunk 2]
```diff
@@ -1944,10 +1944,10 @@
       },
       {
         "groupId": "com.squareup.okhttp3",
-        "artifactId": "okhttp-brotli",
+        "artifactId": "okhttp",
         "version": "4.12.0",
-        "nugetVersion": "4.12.0.2",
-        "nugetId": "Square.OkHttp3.OkHttp.Brotli",
+        "nugetVersion": "4.12.0.3",
```
Comment (jpobst) [inline config.json:1949]: Downgrade from `4.12.0.4`.

File: templates/jakarta/External-Dependency-Info.txt [hunk 3]
```diff
@@ -0,0 +1,282 @@
...
+Do not translate or localize
+
+Xamarin Components for Google Google Guice Dependency Injection Library/Framewrok 
+third party material from the projects listed below. The original copyright 
+notice and the license under which Microsoft received such third party 
+material are set forth below.  Microsoft reserves all other rights not 
+expressly granted, whether by implication, estoppel or otherwise.
+
+########################################
+# Google Guice
```
Comment (jpobst) [inline templates/jakarta/External-Dependency-Info.txt:11]: This file should be Jakarta.

File: templates/jakarta/Project.cshtml [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+  <PropertyGroup>
+    <AndroidClassParser>class-parse</AndroidClassParser>
+    <AndroidCodegenTarget>XAJavaInterop1</AndroidCodegenTarget>
+  </PropertyGroup>
+
+  <PropertyGroup>
+    <Authors>Microsoft</Authors>
+    <Owners>Microsoft</Owners>
+    <Copyright>© Microsoft Corporation. All rights reserved.</Copyright>
+    <PackageProjectUrl>https://go.microsoft.com/fwlink/?linkid=865435</PackageProjectUrl>
```
Comment (jpobst) [inline templates/jakarta/Project.cshtml:26]: https://aka.ms/googleplayservices

File: config.json (hunk 1 above)
Comment (jpobst) [reply config.json:1967]: Still a downgrade from existing `4.12.0.2`.

**Review by jpobst — APPROVED** [review]

**Pull request merged**